website/pages/en/developing/substreams-powered-subgraphs-faq.mdx

```diff
@@ -0,0 +1,89 @@
+
+## What are the benefits of Substreams?
+
+There are many benefits to using Substreams, including:
+
+- Composable: You can stack Substreams modules like LEGO blocks, and build upon community modules, further refining public data.
+
+- High-performance indexing: Orders of magnitude faster indexing through large-scale clusters of parallel operations (think BigQuery).
+
+- Sink anywhere: Sink your data to anywhere you want: PostgreSQL, MongoDB, Kafka, subgraphs, flat files, Google Sheets.
+
+- Programmable: Use code to customize extraction, do transformation-time aggregations, and model your output for multiple sinks.
+
+- All the benefits of the Firehose.
+
+## What is the Firehose?
+
+Developed by [StreamingFast](https://www.streamingfast.io/), the Firehose is a blockchain data extraction layer designed from scratch to process the full history of blockchains at speeds that were previously unseen. Providing a files-based and streaming-first approach, it is a core component of StreamingFast's suite of open-source technologies and the foundation for Substreams.
+
+Go to the [documentation](https://firehose.streamingfast.io/) to learn more about the Firehose.
+
+## What are the benefits of the Firehose?
+
+There are many benefits to using Firehose, including:
+
+- Lowest latency & no polling: In a streaming-first fashion, the Firehose nodes are designed to race to push out the block data first.
+
+- Prevents downtimes: Designed from the ground up for High Availability.
+
+- Never miss a beat: The Firehose stream cursor is designed to handle forks and to continue where you left off in any condition.
+
+- Richest data model:  Best data model that includes the balance changes, the full call tree, internal transactions, logs, storage changes, gas costs, and more.
+
+- Leverages flat files: Blockchain data is extracted into flat files, the cheapest and most optimized computing resource available.
+
+## Where can devs access more information about Substreams-powered subgraphs and Substreams?
+
+The [Substreams documentation](https://substreams.streamingfast.io) will teach you how to do Substreams programming.
+
+The [Substreams-powered subgraphs documentation](../cookbook/substreams-powered-subgraphs/) will show you how to package them for deployment on The Graph.
+
+## What is the role of Rust modules in Substreams?
+
+Rust modules are the equivalent of the AssemblyScript mappers in subgraphs. They are compiled to WASM in a similar way, but the programming model allows for parallel execution. They define the sort of transformations and aggregations you want to apply to the raw blockchain data.
+
+See [modules documentation](https://substreams.streamingfast.io/developers-guide/modules) for details.
```

Comment:
```suggestion
See [modules documentation](/substreams/developers-guide/modules/types/) for details.
```

I believe we should have a [`/substreams/developers-guide/modules/README`](https://github.com/streamingfast/substreams/blob/develop/docs/developers-guide/modules/README.md) route but it is currently a 404. @saihaj and @B2o5T, looks like [it is explicitly filtered out](https://github.com/graphprotocol/docs/blob/main/website/scripts/fetch-remote-filepaths.ts#L52)... could it be because the content is the same as the [`/substreams/concepts-and-fundamentals/modules`](https://github.com/streamingfast/substreams/blob/develop/docs/concepts-and-fundamentals/modules.md) page? Is that intentional or an oversight? Maybe we should link to that instead?

Reply:
I think this was intentional cause of similar content. But good idea that we add a redirect

Reply:
Not similar though, it's exactly the same haha. I meant more, is it intentional that these two (distinct) pages from the Substreams docs have the same content? Maybe one page can be removed.